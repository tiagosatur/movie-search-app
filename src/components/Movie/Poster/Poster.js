import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faFileImage } from '@fortawesome/free-regular-svg-icons'

const Poster = props => {
    const {Title, Poster} = props;
    return (
        <>
            {
                Poster !== 'N/A' 
                ? <img src={Poster} alt={Title} /> 
                : <div>
                    <FontAwesomeIcon icon={faFileImage} style={{fontSize: '100px'}} />
                    <span>Image unnavailable :(</span> 
                </div>
            }
        </>
    );
}

export default Poster;