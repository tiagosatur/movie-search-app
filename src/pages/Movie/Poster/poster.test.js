import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Poster from './Poster';

describe('Poster component test', () => {
    const wrapper =  shallow(
        <Poster />
    );

    it('should call App snapshot correctly', () => {
        const tree = renderer
            .create(wrapper)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})


