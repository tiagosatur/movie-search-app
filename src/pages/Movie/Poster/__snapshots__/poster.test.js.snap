// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster component test should call App snapshot correctly 1`] = `<img />`;
